code/core/src/common/js-package-manager/Yarn1Proxy.ts

```diff
@@ -74,13 +74,11 @@ export class Yarn1Proxy extends JsPackageManager {
     packageName: string,
     basePath = this.cwd
   ): Promise<PackageJson | null> {
-    const packageJsonPath = await findUp(
-      (dir) => {
-        const possiblePath = join(dir, 'node_modules', packageName, 'package.json');
-        return existsSync(possiblePath) ? possiblePath : undefined;
-      },
-      { cwd: basePath }
-    );
+    const dirs = walk.up(basePath ?? '.');
```

Comment:
and again
https://github.com/storybookjs/storybook/pull/31338/files#r2075527459